### PR TITLE
684 issue details fetch data from all active resources

### PR DESCRIPTION
**Pull request by GeorgeGoodall-GovUk**
## What type of PR is this? (check all applicable)

- [x] Refactor
- [x] Feature

## Description
Update the issue details page to two other files: entity issue details and entry issue details. 
getting the correct relevent issues from the entity issue links
also refactored alot of code from the middlewares into common middleware

## Related Tickets & Documents

- Ticket Link #684 

## QA Instructions, Screenshots, Recordings
I recommend looking through the datasets on lambeth and local-authority:WRT 

### Before

_Before screenshot here_

### After

_After screenshot here_

## Added/updated tests?
- [x] Yes
- [ ] No, and this is why: _Please replace this line with details on why tests have not been included_
- [ ] I need help with writing tests

## QA sign off
- [x] Code has been checked and approved
- [x] Design has been checked and approved
- [x] Product and business logic has been checked and proved

## Release Notes

- **New Features**
	- Introduced new middleware functions for handling entity and entry issue details, enhancing data processing capabilities.
	- Added support for pagination and error summary handling in middleware.
	- New HTML formatting function for error messages to improve error display.
	- Updated pagination display in HTML templates to reflect correct row counts.

- **Bug Fixes**
	- Removed outdated middleware functions to streamline request handling.

- **Documentation**
	- Updated routing definitions to reflect new middleware usage.

- **Tests**
	- Expanded test coverage for new and existing middleware functions to ensure robustness.

### CODE REVIEW

**Review by DilwoarH — CHANGES_REQUESTED** [review]

File: src/middleware/common.middleware.js
Comment (GeorgeGoodall-GovUk) [reply src/middleware/common.middleware.js:567]: yep, valibot takes care of that now and sets a default of 0 (as long as your getting it from the parsedParams)

**Review by DilwoarH — DISMISSED** [review]
Approved, please merge after design sign off

**Pull request merged**
